### ISSUE TITLE
Custom Padding When Applied Without Responsive Editing Should Still Apply To All Breakpoints

### ISSUE THREAD

**Issue by timayhelen** [elegantthemes/Divi-Beta#905]
Currently if you edit the padding for any row or section, the custom padding is only applied to desktop. I believe the correct behavior is that any custom padding input should be applied to all three breakpoints unless responsive editing is enabled, in which case we assign the specific values to each breakpoint. 

## Attached PR
- https://github.com/elegantthemes/submodule-builder/pull/1565

**Comment by timayhelen** [comment]
Just an update, this seems to be isolated to certain layouts. It doesn't occur in all situations. You can see the problem in the Divi Builder demo, however:

https://www.elegantthemesdemo.com/?et_fb=1

![row-padding](https://cloud.githubusercontent.com/assets/3333028/21914606/396a66f0-d8e9-11e6-89d5-6a4fdd052be2.gif)

You can download the layout from the demo website to test it out locally.

**Comment by timayhelen** [comment]
I found the cause of this. Basically whenever padding_mobile=\"off\"  or column_padding_mobile=\"off\" for a row or column, then customized padding is only applied to desktop. @fikrirasyid Was this intentional?

**Comment by timayhelen** [comment]
@fikrirasyid Ah I remember now, this is that legacy setting that was hidden. I guess it doesn't play quite right with the responsive editing settings for padding.

**Comment by fikrirasyid** [comment]
@timayhelen I'm currently cross-check it with the PR where the change is occur. In general, what you mentioned above is correct. We actually have a mechanism for phasing out the legacy `padding_mobile` and `column_padding_mobile` attribute, which is whenever responsive mode of the input is triggered (since it is properly imply the intention of the action, which `uses responsive mode` === `want to edit responsive output of the element` === `ditch the legacy setting`) at https://github.com/elegantthemes/submodule-builder/pull/1345/files#diff-7168f624ecf077beb1b5491122b0b258R155

However, it doesn't consider dragging the padding into account, thus the issue.

I'm still thinking through this myself. The most possible way is:

1. When the setting modal is opened, if `padding_mobile` is set to off, update the value to `desktop:updated|tablet:default|mobile:default|responsive:on`, then remove the padding_mobile attribute
2. Make the padding space properly reflect the setting value

 🤔

**Comment by timayhelen** [comment]
> However, it doesn't consider dragging the padding into account, thus the issue.

@fikrirasyid I don't think so. If you enter in the padding into the settings modal the same problem occurs. Whenever padding_mobile="off", padding is only applied to desktop and not to tablet or smartphone.

![row-padding](https://cloud.githubusercontent.com/assets/3333028/21921169/eaba886e-d91a-11e6-86ab-a04138fd15ba.gif)

**Comment by timayhelen** [comment]
@fikrirasyid I guess it's quite confusing to users, because new rows and sections are treated differently than legacy rows and sections. Basically:

- Legacy Rows and Sections with the hidden attribute padding_mobile="off" prevent general padding adjustments from being applied to tablet and smartphone resolutions. This is intended because, if we applied these values and ignored the old settings, lots of websites would have their mobile paddings changed without the user's consent.
- However, new rows and sections that are created without the legacy attribute padding_mobile="off" work different. They applied the general padding to all three breakpoints unless responsive editing has been enabled and custom values have been assigned. 
- Thus it's possible two have a mixture of new and old rows and sections on the same page that work completely differently when adjusting padding, and there is no way for the user to figure out why. 

I think the only solution to this would be to:

- Respect the padding_mobile="off" settings like we do now for all legacy rows and sections. 
- Once a padding value is adjusted for any row or section that contains the padding_mobile="off" attribute, remove the attribute and allow the padding to be applied to all breakpoints just like current rows and sections. 

This way, no old layouts are broken and adjustments to new and old rows and sections will work the same way.

**Comment by fikrirasyid** [comment]
@timayhelen there is a possibility of creating confusion by doing the following:

> Once a padding value is adjusted for any row or section that contains the padding_mobile="off" attribute, remove the attribute and allow the padding to be applied to all breakpoints just like current rows and sections.

By removing `padding_mobile="off"` when adjusted any padding value of row or section, it'll automatically apply responsive view to the padding. For example, let's say we have this configuration:

```
custom_padding: 100|100|100|100
padding_mobile: off
```

This means:

desktop padding: 100px on all corners
tablet padding: default
mobile padding: default

By removing the `padding_mobile="off"` when modifying any padding value (including modifying desktop input only), this scenario can occur:

1. Modifying desktop padding above, let's say by dragging the padding UI
2. `padding_mobile="off"` is removed
3. Suddenly, user will have the following configuration because without explicit responsive mode being active, desktop value is inherited to its tablet and mobile UI:

desktop padding: 100px on all corners
tablet padding: 100px on all corners
mobile padding: 100px on all corners

***

I think we can create a smoother experience by doing a little bit adjustment:

1. The `padding_mobile="off"` is actually has been taken care of pretty smoothly on settings-modal. `padding_mobile` attribute will be removed if the responsive menu is activated (clicking the phone icon)
2. Drop the `padding_mobile` when dragging the padding UI **IF** user is currently on tablet / phone preview screen because it implies the intent of modifying the UI on responsive view.

**Comment by timayhelen** [comment]
> This means:
> 
> desktop padding: 100px on all corners
> tablet padding: default
> mobile padding: default
> 
> By removing the padding_mobile="off" when modifying any padding value (including modifying desktop input only), this scenario can occur:
> 
> Modifying desktop padding above, let's say by dragging the padding UI
> padding_mobile="off" is removed
> Suddenly, user will have the following configuration because without explicit responsive mode being active, desktop value is inherited to its tablet and mobile UI:
> desktop padding: 100px on all corners
> tablet padding: 100px on all corners
> mobile padding: 100px on all corners

Yes, but that is exactly how things work right now for new rows and sections. The desktop value is applied outside the breakpoints and is therefore applied to everything. This is why padding_mobile="off" needs to be removed whenever the desktop value is modified. I think if we just removed padding_mobile="off" whenever padding is adjusted, then all rows and sections will work the same way.

**Comment by fikrirasyid** [comment]
@timayhelen Ah I see, it is pretty tricky to grasp the idea but I think this really helps me understanding what you're suggesting:

> Yes, but that is exactly how things work right now for new rows and sections. The desktop value is applied outside the breakpoints and is therefore applied to everything. This is why padding_mobile="off" needs to be removed whenever the desktop value is modified. I think if we just removed padding_mobile="off" whenever padding is adjusted, then all rows and sections will work the same way.

Alright, to sum it up, what needs to be further adjusted are:

> Whenever padding value is updated, either in VB or BB, on *any* preview mode (zoom, desktop, tablet, phone), using settings-modal input fields or padding drag UI, we need to remove the `padding_mobile` attribute.

Is this correct?

**Comment by timayhelen** [comment]
@fikrirasyid Yeah I think that makes most sense, if I am understanding it correctly.

**Comment by fikrirasyid** [comment]
@timayhelen I'm doing another test to make sure of it, and I think we need one more adjustment. In general, removing `padding_mobile` attribute when any padding value is ok. However to retain user layout when we're removing the `padding_mobile` needs an update. Take a look at these images:

![screen shot 2017-01-14 at 8 44 41 am](https://cloud.githubusercontent.com/assets/916442/21951422/a09bfabe-da38-11e6-9f72-5d29d77e099f.png)

![padding_mobile-test-2](https://cloud.githubusercontent.com/assets/916442/21951442/08f9d428-da39-11e6-8bb2-6392d84f68c6.jpg)

You can test it using this layout: 
[905 check.json.zip](https://github.com/elegantthemes/Divi-Beta/files/705690/905.check.json.zip)

Basically, the equations are:

`padding_mobile = on` === `new responsive_mode = on`

`padding_mobile = off` === `new responsive_mode = on + default_value for tablet & phone inserted if these values are empty`

The difference with what you suggested earlier are:

1. `padding_mobile = off` !== `new responsive_mode = off`, because in `new responsive_mode = off`, phone and tablet inherit desktop value

What do you think?

**Comment by timayhelen** [comment]
@fikrirasyid We just want an old row, once edited, to work exactly like a new row. When you edit the padding value of new rows, that custom padding is placed outside of all breakpoints and applies to the entire page no matter what. We want that same behavior to occur once an old row has been edited, and it seems like removing the padding_mobile attribute would do the trick.

**Comment by lots0logs** [comment]
![image](https://cloud.githubusercontent.com/assets/4675662/21951702/59d1ec6e-d9d0-11e6-9b63-6a01db602c3f.png)

:grin:

**Comment by fikrirasyid** [comment]
@timayhelen I see. So I thought beside retain the behaviour, we need to retain the output to be identical as well. Alright then, removing `padding_mobile` would do the job 👍

**Comment by fikrirasyid** [comment]
@timayhelen @lots0logs I've updated the branch to match your suggestion at https://github.com/elegantthemes/Divi-Beta/issues/905#issuecomment-272595465. Basically, whenever padding value is updated, either from BB, VB, or its draggable padding UI, `padding_mobile` legacy attribute will be deleted.

Note: column's `column_padding_mobile` attribute is located on its row.
